### PR TITLE
66 refactor sfml into raylib

### PR DESCRIPTION
**Pull request by Rafouzou**
- All the display now runs on Raylib
- There is music
- There is a 'settings' page to change the volume of the music
- The game displays the player and one enemy with a parallax behind
- Some new assets were added

### CODE REVIEW

File: src/client/CMakeLists.txt [hunk 1]
```diff
@@ -3,9 +3,22 @@ set(SRCROOT ${PROJECT_SOURCE_DIR}/src/client)
...
+
+FetchContent_Declare(
+    raylib
+    GIT_REPOSITORY "https://github.com/raysan5/raylib.git"
+    GIT_TAG "master"
+    GIT_PROGRESS TRUE
+)
+
+FetchContent_MakeAvailable(raylib)
+
```
Comment (francoisbaudouin) [inline src/client/CMakeLists.txt:19]: remove fetchContent, it's NOT matched in our packages architecture 

add:

find_package(raylib REQUIRED COMPONENTS graphics window system audio)

File: src/client/CMakeLists.txt [hunk 2]
```diff
@@ -42,17 +55,18 @@ add_executable(${TARGET_NAME} ${SRCROOT}/Main.cpp ${SRC})
 # Set the output name for the target binary
 set_target_properties(${TARGET_NAME} PROPERTIES OUTPUT_NAME "r-type_client")
 
-target_link_libraries(${TARGET_NAME} PUBLIC Threads::Threads dl)
+target_link_libraries(${TARGET_NAME} PUBLIC Threads::Threads pthread dl)
 target_link_libraries(${TARGET_NAME} PUBLIC boost::boost)
-target_link_libraries(${TARGET_NAME} PUBLIC sfml::sfml)
+# target_link_libraries(${TARGET_NAME} PUBLIC sfml::sfml)
```
Comment (francoisbaudouin) [inline src/client/CMakeLists.txt:60]: Remove comment. Sfml is not needed anymore

File: src/client/CMakeLists.txt [hunk 3]
```diff
@@ -42,17 +55,18 @@ add_executable(${TARGET_NAME} ${SRCROOT}/Main.cpp ${SRC})
...
+target_link_libraries(${TARGET_NAME} PUBLIC raylib)
 
-target_link_libraries(${TARGET_NAME} PRIVATE Threads::Threads dl)
+target_link_libraries(${TARGET_NAME} PRIVATE Threads::Threads pthread dl)
```
Comment (francoisbaudouin) [inline src/client/CMakeLists.txt:63]: Remove this line, is duplicate from line 58

File: src/client/CMakeLists.txt [hunk 4]
```diff
@@ -42,17 +55,18 @@ add_executable(${TARGET_NAME} ${SRCROOT}/Main.cpp ${SRC})
 # Set the output name for the target binary
 set_target_properties(${TARGET_NAME} PROPERTIES OUTPUT_NAME "r-type_client")
 
-target_link_libraries(${TARGET_NAME} PUBLIC Threads::Threads dl)
+target_link_libraries(${TARGET_NAME} PUBLIC Threads::Threads pthread dl)
 target_link_libraries(${TARGET_NAME} PUBLIC boost::boost)
```
Comment (francoisbaudouin) [inline src/client/CMakeLists.txt:59]: remove the line and move boost::boost to the line 66

the line become
target_link_libraries(${TARGET_NAME} PRIVATE Boost::filesystem Boost::system Boost::serialization, boost::boost)

**Review by francoisbaudouin — CHANGES_REQUESTED** [review]
See comment.

Be aware, don't duplicate code and respect the existent packet manager.
If you have question of packet manager ask me.

Also, pls lint the code. To do that follow the next instruction:

under fedora:

sudo dnf install clang-tools-extra

ubuntu:

sudo apt install clang-format

In vs code, install extension Clang-Format by publisher:"Xaver Hellauer"

**Review by ThibaultGuyony — APPROVED** [review]
LGTM

**Review by francoisbaudouin — APPROVED** [review]
ok

**Pull request merged**